backend/internal/userschema/model/userschema.go

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package model defines the data structures for user schema management.
+package model
+
+import (
+	"encoding/json"
+)
+
+// UserSchema represents a user type schema definition.
+type UserSchema struct {
+	ID     string          `json:"id,omitempty"`
+	Name   string          `json:"name,omitempty"`
+	Schema json.RawMessage `json:"schema,omitempty"`
+}
+
+// UserSchemaListItem represents a simplified user schema for listing operations.
+type UserSchemaListItem struct {
+	ID   string `json:"id,omitempty"`
+	Name string `json:"name,omitempty"`
+}
+
+// Link represents a hypermedia link in the API response.
+type Link struct {
```

Comment:
Suggestion:
Since this is reused in many APIs, better to define a single constant in system/constants.
We can address this in a separate PR